Tests/SwiftParserTest/translated/TryTests.swift

```diff
@@ -204,98 +188,76 @@ final class TryTests: XCTestCase {
+        DiagnosticSpec(locationMarker: "3️⃣", message: "'try' cannot be used with 'func'"),
+      ], fixedSource: """
+      class TryDecl {
+        let singleLet = try foo()
+        var singleVar = try foo()
+        try
+        func method() {}
+      }
+      """
     )
   }
 
   func testTry12a() {
     AssertParse(
       """
-      func test() throws -> Int {
-        try1️⃣ 2️⃣while true {
-          try3️⃣ 4️⃣break
-        }
+      1️⃣try while true {
+        2️⃣try break
       }
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 2: 'try' cannot be used with 'while'
-        DiagnosticSpec(locationMarker: "1️⃣", message: "consecutive statements on a line must be separated by ';'"),
-        DiagnosticSpec(locationMarker: "2️⃣", message: "expected expression in 'try' expression"),
-        // TODO: Old parser expected error on line 3: 'try' cannot be used with 'break'
-        DiagnosticSpec(locationMarker: "3️⃣", message: "consecutive statements on a line must be separated by ';'"),
-        DiagnosticSpec(locationMarker: "4️⃣", message: "expected expression in 'try' expression"),
+        DiagnosticSpec(locationMarker: "1️⃣", message: "'try' cannot be used with 'while'"),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "'try' cannot be used with 'break'"),
       ]
     )
   }
 
   func testTry12b() {
     AssertParse(
       """
-      func test() throws -> Int {
-        try1️⃣ 2️⃣throw 3️⃣
-      }
+      1️⃣try throw 2️⃣
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 5: 'try' must be placed on the thrown expression, Fix-It replacements: 3 - 7 = '', 3 - 3 = 'try '
-        // TODO: Old parser expected error on line 5: expected expression in 'throw' statement
-        DiagnosticSpec(locationMarker: "1️⃣", message: "consecutive statements on a line must be separated by ';'"),
+        DiagnosticSpec(locationMarker: "1️⃣", message: "'try' must be placed on the thrown expression", fixIts: ["move 'try' after 'throw'"]),
         DiagnosticSpec(locationMarker: "2️⃣", message: "expected expression in 'try' expression"),
-        DiagnosticSpec(locationMarker: "3️⃣", message: "expected expression in 'throw' statement"),
-      ]
+      ], fixedSource: "throw try <#expression#>"
     )
   }
 
   func testTry12c() {
     AssertParse(
       """
-      func test() throws -> Int {
-        try1️⃣ 2️⃣return
-      }
+      1️⃣try return
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 7: 'try' cannot be used with 'return'
-        DiagnosticSpec(locationMarker: "1️⃣", message: "consecutive statements on a line must be separated by ';'"),
-        DiagnosticSpec(locationMarker: "2️⃣", message: "expected expression in 'try' expression"),
+        DiagnosticSpec(message: "'try' cannot be used with 'return'"),
```

Comment:
Hmmm... I'm not really sure what I'd expect here. This is what we currently do so I suppose it's fine.

This is similar to the `try while` case above. Having it expect `return try` and `while try` sort of makes more sense to me since if someone has written `try` they're probably going to call something that throws. I don't feel very strongly about this though, pretty edge case and removing also seems fine.

Reply:
Yeah, not sure if it's really worth having a diagnostic for this but since we have it now, I don't think it does any harm.